modules/platforms/dotnet/Apache.Ignite.Core/Impl/Cache/CacheImpl.cs

```diff
@@ -981,23 +992,39 @@ private Task<long> SizeAsync0(int? part, params CachePeekMode[] modes)
             var holder = new CacheEntryProcessorHolder(processor, arg,
                 (e, a) => processor.Process((IMutableCacheEntry<TK, TV>)e, (TArg)a), typeof(TK), typeof(TV));
 
-            return DoOutOpAsync(CacheOp.InvokeAsync, writer =>
-                {
-                    writer.WriteObjectDetached(key);
-                    writer.WriteObjectDetached(holder);
-                },
-                r =>
-                {
-                    if (r == null)
-                        return default(TRes);
+            var ptr = AllocateNoTx(holder);
```

Comment:
Can you describe this a little bit, please. It's kinda confusing to read  `AllocateNoTx` right after `StartTx`. Would not `AllocateNoTx` always return 0? Cause at first glance the original issue was caused by PlatformsUtils.readSet method.

Reply:
I've updated the PR description (so this gets into the commit message too).

`StartTx` starts a transaction only when there is a `TransactionScope` and the cache is `Transactional`, in all other cases it just does nothing. `AllocateNoTx` checks if there is an active Ignite transaction, which can be created in two ways - with a `TransactionScope` or with `ITransactions` API.

I agree that naming is confusing here, let me try to improve it.

Reply:
Changed names to `StartTxIfNeeded` and `AllocateIfNoTx`, hopefully this is less confusing.